patch.crates-io.sh

```diff
@@ -26,6 +43,12 @@ cd "$(dirname "$0")"
 source "$solana_dir"/scripts/read-cargo-variable.sh
 solana_ver=$(readCargoVariable version "$solana_dir"/sdk/Cargo.toml)
 
+if $maybe_patch_zk_token_sdk; then
+  maybe_patch_zk_token_sdk="solana-zk-token-sdk = { path = \"$solana_dir/zk-token-sdk\" }"
+else
+  maybe_patch_zk_token_sdk=
+fi
```

Comment:
This special casing doesn't feel the best (even though we are giving zk-token-sdk special handling on crates.io, etc). Could we do something clever and build a list of all the dependencies and their relative dirs, and add each path patch only if the dir exists?

Reply:
I have a dangling empty dir when I switched branches.  I know we clean up after the target dir, but I think CI may have the same issue

Reply:
Oh :(

Reply:
Could check for Cargo.toml inside by the looks

Reply:
ya, checking for Cargo.toml should avoid the need for `--no-patch-zk-token-sdk`:
```
if [[ -r "$solana_dir"/zk-token-sdk/Cargo.toml ]]; then
  ..
fi
```

Reply:
in https://github.com/solana-labs/solana-program-library/pull/2969/commits/0ee8d83a2b712b617f42baecf41c96404a158312